### ISSUE TITLE
In MappingListModel and related classes, use of "Qt::CheckStateRole + xyz" is not appropriate

### ISSUE THREAD

**Issue by sofian** [mapmapteam/mapmap#234]
Both "solo" and "locked" state are controlled by looking at role "Qt::CheckStateRole + <a certain value>" but this is not appropriate. I think the right approach is rather to use the index.column() option. However, as it is now, the code "works" but it is a "hack". Therefore we should fix it.

**Comment by baydam** [comment]
Yes we should fix it and I agree with that but I think that if Qt here decided to jump between `Qt::CheckStateRole = 10` and `Qt::InitialSortOrderRole = 14` is because it allows to use the 11 12 and 13 numbers.

**Comment by sofian** [comment]
All right in that case this is really low priority (probably wontfix actually). Let's leave it there but outside of milestones.
